src/agent.ts

```diff
@@ -175,7 +178,8 @@ export default class SocksProxyAgent extends Agent {
 			return tls.connect({
 				...omit(opts, 'host', 'hostname', 'path', 'port'),
 				socket,
-				servername
+				servername,
+				rejectUnauthorized: this.tlsRejectUnauthorized
```

Comment:
although this PR is fixing the original issue, do you think a more complete solution could be to just pass all the rest props as part of `tls.connect` options?

something like:

```
return tls.connect({
				...omit(opts, 'host', 'hostname', 'path', 'port'),
				socket,
				servername,
                ...this.tlsOpts
```

since eventually people want to customize tls connection details, not just `rejectUnauthorized`

Reply:
hey, @Kikobeats! I thought about it, maybe we can let them override everything with `tls.connect` options. will fix it.

Reply:
can you update the PR then? 🙂

Reply:
@Kikobeats done